tools/gen_compdb.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# Copyright 2016-2019 Envoy Project Authors
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+import shlex
+import subprocess
+from pathlib import Path
+
+# This is copied from https://github.com/envoyproxy/envoy and remove unnecessary code.
+
+# This method is equivalent to https://github.com/grailbio/bazel-compilation-database/blob/master/generate.py
+def generate_compilation_database(args):
+    # We need to download all remote outputs for generated source code. This option lives here to override those
+    # specified in bazelrc.
+    bazel_startup_options = shlex.split(os.environ.get("BAZEL_STARTUP_OPTION_LIST", ""))
+    bazel_options = shlex.split(os.environ.get("BAZEL_BUILD_OPTION_LIST", "")) + [
+        "--remote_download_outputs=all",
+    ]
+
+    source_dir_targets = args.bazel_targets
+
+    subprocess.check_call(["bazel", *bazel_startup_options, "build"] + bazel_options + [
```

Comment:
Can the name for the bazel binary be obtained from an environment variable or flag (with default "bazel"), to accomodate developers who use bazelisk?

Reply:
Actually, I don't think we should do it in the script. (I also updated the envoy's script to remove the bazel alias support)

Basically, I think this should be part of the base development env. For example, the homebrew will link the bazelisk with name `bazel` directly. (And I think this is also the correct way to use bazelisk). Or the caller could create a temporary alias by `alias bazel=bazelisk`.

I will prefer to let the script to focus on the it's core target.

Reply:
Fair enough, SGTM!